ethereum/core/src/integration-test/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContractIntegrationTest.java

```diff
@@ -171,4 +170,36 @@ public void testSendAndReceive() throws IOException {
 
     assertThat(actual).isEqualTo(BytesValue.fromHexString(DEFAULT_OUTPUT));
   }
+
+  @Test
+  public void testNoPrivateKeyError() throws RuntimeException {
+    List<String> publicKeys = testHarness.getPublicKeys();
+    publicKeys.add("noPrivateKey");
+
+    String s = new String(VALID_PRIVATE_TRANSACTION_RLP_BASE64, UTF_8);
+    SendRequest sc = new SendRequest(s, publicKeys.get(0), publicKeys);
+    try {
+      enclave.send(sc);
+    } catch (IOException e) {
+      if (!e.getMessage().contains("EnclaveDecodePublicKey")) {
```

Comment:
Test errors with `@Test` annotation - https://junit.org/junit4/faq.html#atests_7

Reply:
I want to test the specific error message and not just if the exception is throw.

Reply:
I don't agree. This should return an ErrorResponse object. Any exception should be related to connection issues. If the `send()` returns a response (even if it is an error) I think it should be kept in an `ErrorResponse` object